bundler/spec/dependabot/bundler/file_updater_spec.rb

```diff
@@ -1596,6 +1596,33 @@
       end
     end
 
+    context "for a gem that depends on bundler" do
+      subject(:updated_gemfile) do
+        updated_files.find { |f| f.name == "Gemfile" }
+      end
+
+      let(:dependency_files) { bundler_project_dependency_files("guard_bundler") }
+      let(:dependency_name) { "guard-bundler" }
+      let(:dependency_version) { "3.0.0" }
+      let(:dependency_previous_version) { "2.2.1" }
+      let(:requirements) do
+        [{
+          file: "Gemfile",
+          requirement: "~> 2.2.1",
+          groups: [],
+          source: nil
+        }]
+      end
+
+      it "raises an error", :bundler_v1_only do
+        expect { updated_gemfile.content }.to raise_error(/Bundler could not find compatible versions for gem "bundler"/)
```

Comment:
Nit: calling a method on a raise feels a bit weird but not blocking 

```suggestion
        expect { updated_gemfile }.to raise_error(/Bundler could not find compatible versions for gem "bundler"/)
```